core/referral/engine.go

```diff
@@ -484,95 +459,85 @@ func (e *Engine) computeReferralSetsStats(ctx context.Context, epoch types.Epoch
 		e.referralSetsNotionalVolumes.Add(epoch.Seq, setID, volumeForEpoch)
 	}
 
-	partiesTakerVolume := map[types.PartyID]*num.Uint{}
+	takerVolumePerReferee := map[types.PartyID]*num.Uint{}
 
 	for partyID, setID := range e.referees {
 		volumeForEpoch := e.marketActivityTracker.NotionalTakerVolumeForParty(string(partyID))
 		e.referralSetsNotionalVolumes.Add(epoch.Seq, setID, volumeForEpoch)
-		partiesTakerVolume[partyID] = volumeForEpoch
+		takerVolumePerReferee[partyID] = volumeForEpoch
 	}
 
 	if e.programHasEnded {
 		return
 	}
 
-	e.computeFactorsByReferee(ctx, epoch.Seq, partiesTakerVolume)
+	e.computeFactorsByReferee(ctx, epoch.Seq, takerVolumePerReferee)
 }
 
-func (e *Engine) computeFactorsByReferee(ctx context.Context, epoch uint64, takerVolumePerParty map[types.PartyID]*num.Uint) {
+func (e *Engine) computeFactorsByReferee(ctx context.Context, epoch uint64, takerVolumePerReferee map[types.PartyID]*num.Uint) {
 	e.factorsByReferee = map[types.PartyID]*types.RefereeStats{}
 
 	allStats := map[types.ReferralSetID]*types.ReferralSetStats{}
-	tiersLen := len(e.currentProgram.BenefitTiers)
 
 	for setID, set := range e.sets {
 		setStats := &types.ReferralSetStats{
 			AtEpoch:                  epoch,
 			SetID:                    setID,
+			WasEligible:              false,
+			ReferralSetRunningVolume: num.UintZero(),
 			RefereesStats:            map[types.PartyID]*types.RefereeStats{},
-			ReferralSetRunningVolume: e.referralSetsNotionalVolumes.RunningSetVolumeForWindow(setID, e.currentProgram.WindowLength),
 			RewardFactor:             num.DecimalZero(),
+			RewardsMultiplier:        num.DecimalOne(),
+			RewardsFactorMultiplier:  num.DecimalZero(),
 		}
 
-		for i := tiersLen - 1; i >= 0; i-- {
-			tier := e.currentProgram.BenefitTiers[i]
-			if setStats.RewardFactor.IsZero() && setStats.ReferralSetRunningVolume.GTE(tier.MinimumRunningNotionalTakerVolume) {
-				setStats.RewardFactor = tier.ReferralRewardFactor
-				break
-			}
-		}
-
-		balance, _ := e.staking.GetAvailableBalance(set.Referrer.PartyID.String())
+		stakingBalance, _ := e.staking.GetAvailableBalance(set.Referrer.PartyID.String())
+		setStats.WasEligible = stakingBalance.GTE(e.referralProgramMinStakedVegaTokens)
 
-		if balance.GTE(e.referralProgramMinStakedVegaTokens) {
-			setStats.WasEligible = true
+		if setStats.WasEligible {
+			setStats.ReferralSetRunningVolume = e.referralSetsNotionalVolumes.RunningSetVolumeForWindow(setID, e.currentProgram.WindowLength)
+			setStats.RewardFactor = e.matchRewardFactor(setStats.ReferralSetRunningVolume)
+			setStats.RewardsMultiplier = e.matchRewardMultiplier(stakingBalance)
+			setStats.RewardsFactorMultiplier = num.MinD(
+				setStats.RewardFactor.Mul(setStats.RewardsMultiplier),
+				e.referralProgramMaxRewardProportion,
+			)
 		}
```

Comment:
I suppose we should still set the ReferralSetRunningVolume to something, even if not eligible, we'd like to know the running volume.

For the factor and multiplies I'd also set them to defaults values.

Reply:
> For the factor and multiplies I'd also set them to defaults values.

That is done, it's in the struct initializer.